interface/ceed-qfunctioncontext.c

```diff
@@ -399,6 +495,147 @@ int CeedQFunctionContextRestoreData(CeedQFunctionContext ctx, void *data) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Register QFunctionContext a field holding a double precision value
+
+  @param ctx               CeedQFunctionContext
+  @param field_name        Name of field to register
+  @param field_offset      Offset of field to register
+  @param field_description Description of field, or NULL for none
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedQFunctionContextRegisterDouble(CeedQFunctionContext ctx,
+                                       const char *field_name, size_t field_offset,
+                                       const char *field_description) {
+  return CeedQFunctionContextRegisterGeneric(ctx, field_name, field_offset,
+         field_description, "double", sizeof(double));
+}
```

Comment:
Do we want this to always be single values or should it be an array (perhaps usually of length 1)? I honestly don't know. What's here will often be sufficient and maybe the ergonomics are worth it.

Reply:
I'll comment more in the morning - I was wondering (honestly not sure)... If we add an array of a fixed length to the struct and then add fields
```
typedef struct {
  double forcing[3],
  double time,
} myContext;
```

Could we just add "forcing_x", "forcing_y", and "forcing_z" as separate setable fields? That feels like it should be legal.

Reply:
We could, but I'm concerned about a higher level library/application writing loops that construct many strings. As an extreme, but not hypothetical example, the context could hold weights of a neural network used by CRIKit to implement the constitutive relation. It'd be nice to provide the hundreds of values in an array rather than lots of string queries.

Reply:
Wouldn't that scenario be better served by using GetContextData though?

Reply:
Perhaps, but that requires knowing the layout of the entire struct, not just one part. For example, suppose one is using a standard model for thermal diffusivity (some function of temperature, thus needing some coefficients), but an NN representation of stress-strain. Access by name seems to make it easier to compose and to set interpretable parameters in an input file while also supporting programmatic access to some less interpretable parameters that you probably won't list out in an input file.

Reply:
We perhaps leave the single variants for now but design the internals to support arrays when we get to a problem where that is helpful?

Reply:
Sounds good to me.